src/goto-programs/graphml_witness.cpp

```diff
@@ -241,10 +241,9 @@ void graphml_witnesst::operator()(const goto_tracet &goto_trace)
       continue;
     }
 
-    goto_tracet::stepst::const_iterator next=it;
-    for(++next;
-        next!=goto_trace.steps.end() &&
-        (step_to_node[next->step_nr]==sink || it->pc==next->pc);
+    auto next = std::next(it);
+    for(; next != goto_trace.steps.end() &&
+          (step_to_node[next->step_nr] == sink || &(*it->pc) == &(*next->pc));
```

Comment:
In line with the earlier comment: equality over iterators is generally useful, thus use the to-be-introduced function.